dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/BanOptionalDependencySelector.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.dependencies;
+
+import org.eclipse.aether.collection.DependencyCollectionContext;
+import org.eclipse.aether.collection.DependencySelector;
+import org.eclipse.aether.graph.Dependency;
+
+/**
+ * A dependency selector that disallows all optional dependencies regardless of depth.
```

Comment:
To follow Maven's default behavior, we need to include "direct optional dependencies" (depth 1).

`org.eclipse.aether.util.graph.selector.OptionalDependencySelector` does that.

Reply:
That would be only for a specific artifact. This is for a lost of artifacts so all optional dependencies are effectively transitive.

Reply:
Per docs,  OptionalDependencySelector "excludes optional dependencies which occur beyond level one of the dependency graph." This need to exclude all optional dependencies, at level one and beyond.

Reply:
Indeed using that instead of this breaks the tests.